kernel/arch/dreamcast/sound/snd_stream.c

```diff
@@ -746,116 +809,14 @@ int snd_stream_poll(snd_stream_hnd_t hnd) {
     }
 
     write_pos = samples_to_bytes(hnd, stream->last_write_pos);
+    got_bytes = snd_stream_fill(hnd, write_pos, needed_bytes);
 
-    if(stream->req_data) {
-        got_bytes = stream->req_data(hnd,
-            (stream->spu_ram_sch[0] | SPU_RAM_UNCACHED_BASE) + write_pos,
-            (stream->channels == 2 ?
-                ((stream->spu_ram_sch[1] | SPU_RAM_UNCACHED_BASE) + write_pos) : 0),
-            needed_bytes);
-
-        if(got_bytes) {
-            if(got_bytes < needed_bytes) {
-                needed_bytes = got_bytes;
-            }
-            needed_samples = bytes_to_samples(hnd, needed_bytes);
-            goto stream_poll_exit;
-        }
-    }
-    if(!stream->get_data) {
+    if(got_bytes <= 0) {
```

Comment:
snd_stream_fill() returns size_t which is always 0+

Reply:
Good catch, I'll fix it.